src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl

```diff
@@ -199,22 +199,40 @@ SubquoModule{MPolyDecRingElem{fpFieldElem, fpMPolyRingElem}}
 ```
 """
 function canonical_bundle(X::AbsProjectiveVariety)
-   Pn = ambient_coordinate_ring(X)
-   n = ngens(Pn)-1
-   I = defining_ideal(X)
-   c = codim(I)
-   A, _ = quo(Pn, I)
-   FA = free_resolution(A, algorithm = :mres)
-   FAC = FA.C
-   r = range(FAC)
-   C = shift(FAC[first(r):-1:1], -c)
-   F = free_module(Pn, 1)
-   OmegaPn = grade(F, [n+1])
-   D = hom(C, OmegaPn);
-   Omega = homology(D, 0);
-   Omega, _ = prune_with_map(Omega);  
-   return Omega
+  Pn = ambient_coordinate_ring(X)
+  n = ngens(Pn)-1
+  I = defining_ideal(X)
+  c = codim(I)
+  A, _ = quo(Pn, I)
+  FA = free_resolution(A, algorithm = :fres)
+  C = Oscar.SimpleComplexWrapper(FA.C[0:first(range(FA.C))])
+  C_simp = simplify(C);
+  C_shift = shift(C_simp, c);
+  OmegaPn = graded_free_module(Pn, [n+1])
+  D = hom(C_shift, OmegaPn);
+  D_simp = simplify(D);
+  #return simplify(homology(D, 0)[1])[1]
```

Comment:
Optional: get rid of the semicolons and the commented out line by applying this patch
```suggestion
  C_simp = simplify(C)
  C_shift = shift(C_simp, c)
  OmegaPn = graded_free_module(Pn, [n+1])
  D = hom(C_shift, OmegaPn)
  D_simp = simplify(D)
```